### PR TITLE
feat: Establish basic network connection

### PR DESCRIPTION
**Pull request by TannerGabriel**
## This PR

- Adds the `socketIOClient`, `okhttp` and `mockk` libraries to the project
- Add basic client for both SocketIO and HTTP

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -11,7 +13,9 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.Wizard"
-        tools:targetApi="31">
+        tools:targetApi="31"
+        android:usesCleartextTraffic="true"
```
Comment (FRALEXAlexander) [inline app/src/main/AndroidManifest.xml:17]: Why is this needed?

File: app/src/main/java/at/aau/serg/MainActivity.kt [hunk 2]
```diff
@@ -5,6 +5,8 @@ import androidx.activity.enableEdgeToEdge
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
+import at.aau.serg.network.SocketHandler
+import java.lang.Thread.sleep
```
Comment (FRALEXAlexander) [inline app/src/main/java/at/aau/serg/MainActivity.kt:9]: SocketHandler and Thread are unused imports

**Review by FRALEXAlexander — CHANGES_REQUESTED** [review]
A base URL could be added for the HTTP calls

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (TannerGabriel) [reply app/src/main/AndroidManifest.xml:17]: As far as I can tell if this is not enabled, we would need to use HTTPS and WSS which the backend doesn't support yet.

**Review by FRALEXAlexander — APPROVED** [review]
LGTM
